sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/placement/MarathonConstraintParser.java

```
@@ -348,24 +387,23 @@ public PlacementRule run(StringMatcher taskFilter, String fieldName, String oper
                         "Unable to parse max parameter as integer for '%s' operation: %s",
                         operatorName, requiredParameter), e);
             }
-            if (isHostname(fieldName)) {
-                return new MaxPerHostnameRule(max, taskFilter);
-            } else {
-                // Ensure that:
-                // - Task sticks to nodes with matching fieldName defined at all (AttributeRule)
-                // - Task doesn't exceed one instance on those nodes (MaxPerAttributeRule)
-                StringMatcher matcher = RegexMatcher.createAttribute(fieldName, ".*");
-                return new AndRule(
-                        AttributeRuleFactory.getInstance()
-                                .require(matcher), new MaxPerAttributeRule(max, matcher, taskFilter));
+
+            switch (PlacementUtils.getField(fieldName)) {
```

should the default case throw an `UnsupportedOperationException`?